Sources/Logging/Logging.swift

```diff
@@ -472,24 +473,30 @@ public struct MultiplexLogHandler: LogHandler {
     public init(_ handlers: [LogHandler]) {
         assert(!handlers.isEmpty)
         self.handlers = handlers
+        self.effectiveLogLevel = MultiplexLogHandler.effectiveLogLevel(handlers.map { $0.logLevel })
     }
 
     public var logLevel: Logger.Level {
         get {
-            return self.handlers[0].logLevel
+            return self.effectiveLogLevel
         }
         set {
             self.mutatingForEachHandler {
                 $0.logLevel = newValue
             }
+            self.effectiveLogLevel = MultiplexLogHandler.effectiveLogLevel(self.handlers.map { $0.logLevel })
         }
     }
 
+    static func effectiveLogLevel(_ levels: [Logger.Level]) -> Logger.Level {
+        return levels.min() ?? .trace
+    }
+
     public func log(level: Logger.Level,
                     message: Logger.Message,
                     metadata: Logger.Metadata?,
                     file: String, function: String, line: UInt) {
-        self.handlers.forEach { handler in
+        for handler in self.handlers where handler.logLevel <= level {
```

Comment:
wouldn't delegating to the underlying `handler.log` already filter by level? what am I missing?

Reply:
See the analysis here: https://github.com/apple/swift-log/pull/140/files#r447320481

Reply:
> the underlying handler.log already filter by level? what am I missing?

No, a handler is not normally performing the log level check -- only the Logger.log() API does that:

```
    @inlinable
    public func log(level: Logger.Level,
                    _ message: @autoclosure () -> Logger.Message,
                    metadata: @autoclosure () -> Logger.Metadata? = nil,
                    file: String = #file, function: String = #function, line: UInt = #line) {
        if self.logLevel <= level {
            self.handler.log(level: level,
                             message: message(),
                             metadata: metadata(),
                             file: file, function: function, line: line)
        }
    }
```

and handlers are naive and "if i am called, it seems I should log." This is why we have to make the `multiplex.logLevel` smarter as we do in this PR:

- so the change to logLevel in this PR causes the `Logger.log()` to dispatch to this multiplex handler when _at least one of the log handlers would like to log this message_
- but since now we don't know which logger that was... we have to check.

There's an open question how `multiplex.logLevel = .info` relates to this all, since then all levels are made equal and all will log. But what about if we only do:

```
logger1 at debug
logger2 at info

multiplex.debug() // we only want it to be logged at debug at logger1, not at logger2 as well
```

Reply:
For context, `StreamLogHandler`:

```swift
    public func log(level: Logger.Level,
                    message: Logger.Message,
                    metadata: Logger.Metadata?,
                    file: String, function: String, line: UInt) {
        let prettyMetadata = metadata?.isEmpty ?? true
            ? self.prettyMetadata
            : self.prettify(self.metadata.merging(metadata!, uniquingKeysWith: { _, new in new }))
```

Reply:
ah you are correct, I forgot the filter was at the Logger level not backend one. this is 100%